Runtime/MeshSimplifier.cs

```diff
@@ -799,6 +812,51 @@ private double VertexError(ref SymmetricMatrix q, double x, double y, double z)
                 + 2 * q.m5 * y * z + 2 * q.m6 * y + q.m7 * z * z + 2 * q.m8 * z + q.m9;
         }
 
+
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private double CurvatureError(Vertex Vi, Vertex Vj)
+        {
+            //System.Diagnostics.Stopwatch w = new System.Diagnostics.Stopwatch();
+            //w.Start();
+
+            double diffVector = (Vi.p - Vj.p).Magnitude;
+
+            HashSet<Triangle> trianglesWithVi = GetTrianglesContainingVertex(Vi);
+            HashSet<Triangle> trianglesWithVj = GetTrianglesContainingVertex(Vj);
+            HashSet<Triangle> trianglesWithViOrVjOrBoth = new HashSet<Triangle>(trianglesWithVi);
+            trianglesWithViOrVjOrBoth.UnionWith(trianglesWithVj);
+            HashSet<Triangle> trianglesWithViAndVjBoth = GetTrianglesContainingBothVertices(Vi, Vj);
+
+
+            double maxDotOuter = 0;
+
+            foreach (var triangleWithViOrVjOrBoth in trianglesWithViOrVjOrBoth)
+            {
+                double maxDotInner = 0;
+
+                Vector3 normVecWithViOrVjOrBoth = GetTriangleNormalVector(triangleWithViOrVjOrBoth);
```

Comment:
The **n** property of the triangles is the normal vector. So you would save a lot of performance by using it instead of recomputing it for the same triangle multiple times.